### ISSUE TITLE
iframe unmount callback

### ISSUE THREAD

**Issue by paztis** [ryanseddon/react-frame-component#178]
I know that this component is really old but it is really correctly working
There's just one missing feature: be able to provide a unmount callback to be able to remove few listeners in parent app.

Is it possible to add it ?

thanks

**Comment by ryanseddon** [comment]
> I know that this component is really old but it is really correctly working

Being old isn't such a bad thing 😄 

What's your use case, do you have a codesandbox or similar I could look at?

**Comment by ryanseddon** [comment]
Good PR opportunity here if you want to contribute

**Comment by Reflex-Gravity** [comment]
Can I take this?

All I need to do is add a new prop `contentWillUnMount` and pass it to `Content` and call it on componentWillUnmount of Content component rigtht?

**Comment by ryanseddon** [comment]
yes please! just make sure you add tests and documentation to cover the new stuff

**Comment by Reflex-Gravity** [comment]
Can you help through the steps to set up the project and run it?

**Comment by ryanseddon** [comment]
Good prompt for me to add a `contributing.md` file!

1. Pull down the repo
2. Run `npm install`
3. Run `npm run serve`
4. Open `http://localhost:8080` 
5. Now any changes you make to the library or example files will rebuild and refresh the browser
6. Tests are run by `npm t`

**Comment by ryanseddon** [comment]
Implemented in upcoming PR.

**Feature:** Added `contentWillUnmount` callback prop that is called when the Frame component is unmounted.

**Usage:**
```jsx
<Frame 
  contentDidMount={() => console.log('mounted')}
  contentWillUnmount={() => console.log('unmounting')}
>
  <YourComponent />
</Frame>
```

**Changes:**
- src/Frame.jsx: Added `contentWillUnmount` prop
- src/Content.jsx: Added `componentWillUnmount` lifecycle method to call the callback
- index.d.ts: Updated TypeScript definitions
- Added test case for the new callback

This allows users to clean up listeners or perform other cleanup actions when the Frame is unmounted.
